src/zenml/model/model_config.py

```diff
@@ -103,8 +274,10 @@ def _create_model_version(
         Returns:
             The model version based on configuration.
         """
-        if self._model_version is not None:
-            return self._model_version
+        if self.model_version is not None:
+            from zenml.models.model_models import ModelVersionResponseModel
+
+            return ModelVersionResponseModel(**dict(self.model_version))
```

Comment:
Same here, isn't `self.model_version` already a `ModelVersionResponseModel`?

Reply:
I'll redesign this

Reply:
I decided to get rid of this caching, as it produces unexpected results and always returns the version cached at the very beginning of the pipeline run.